docs/tutorial/custom-title-bar.md

```diff
@@ -0,0 +1,146 @@
+use the native UI to control the window.
+
+```js
+const { BrowserWindow } = require('electron')
+const win = new BrowserWindow({ titleBarStyle: 'customButtonsOnHover' })
+```
+
+#### Customize the traffic light position _macOS_
+
+To modify the position of the traffic light window controls, there are two configuration
+options available.
+
+Applying `hiddenInset` title bar style will shift the vertical inset of the traffic lights
+by a fixed amount.
+
+```js title='main.js'
+const { BrowserWindow } = require('electron')
+const win = new BrowserWindow({ titleBarStyle: 'hiddenInset' })
+```
+
+If you need more granular control over the positioning of the traffic lights, you can pass
+a set of coordinates to the `trafficLightPosition` option in the `BrowserWindow`
+constructor.
+
+```js title='main.js'
+const { BrowserWindow } = require('electron')
+const win = new BrowserWindow({
+  titleBarStyle: 'hidden',
+  trafficLightPosition: { x: 10, y: 10 }
+})
+```
+
+#### Show and hide the traffic lights programmatically _macOS_
+
+You can also show and hide the traffic lights programmatically from the main process.
+The `win.setWindowButtonVisibility` forces traffic lights to be show or hidden depending
+on the value of its boolean parameter.
+
+```js title='main.js'
+const { BrowserWindow } = require('electron')
+const win = new BrowserWindow()
+// hides the traffic lights
+win.setWindowButtonVisibility(false)
+```
+
+> **Note**
+> Given the number of APIs available, there are many ways of achieving this. For instance,
+> combining `frame: false` with `win.setWindowButtonVisibility(true)` will yield the same
+> layout outcome as setting `titleBarStyle: 'hidden'`.
+
+#### Custom window controls
+
+The [Window Controls Overlay API][] is a web standard that gives web apps the ability to
+customize their title bar region when installed on desktop. Electron exposes this API
+through the `titleBarOverlay` option in the `BrowserWindow` constructor. When `titleBarOverlay` is enabled, the window controls become exposed in their default
+position, and DOM elements cannot use the area underneath this region.
+
+> **Note**
+> `titleBarOverlay` requires the `titleBarStyle` param in the `BrowserWindow` constructor to have a value other than `default`.
+
+The custom title bar tutorial covers a [basic example][Add native window controls] of exposing window controls by setting `titleBarOverlay: true`. The height, color _Windows_ _Linux_, and symbol colors _Windows_ of the window controls can be customized further by setting `titleBarOverlay` to an object.
+
+The value passed to the `height` property must be an integer. The `color` and `symbolColor` properties accept `rgba()`, `hsla()`, and `#RRGGBBAA` color formats and support transparency. If a color option is not specified, the color will default to its system color for the window control buttons. Similarly, if the height option is not specified it will default to the default height:
```

Comment:
> Similarly, if the height option is not specified it will default to the default height

Would be better to add a bit more detail on what the default height is